meteor/client/ui/Settings/ShowStyle/VariantSettings.tsx

```diff
@@ -43,9 +71,56 @@ export const ShowStyleVariantsSettings = withTranslation()(
 			this.state = {
 				editedMappings: [],
 				timestampedFileKey: Date.now(),
+				dndVariants: this.getOrderedShowStyleVariants(),
 			}
 		}
 
+		componentDidUpdate(
+			prevProps: Readonly<Translated<IShowStyleVariantsProps>>,
+			prevState: Readonly<IShowStyleVariantsSettingsState>
+		) {
+			if (this.showStyleVariantsChanged(prevState, prevProps)) {
+				timer = setTimeout(
+					() =>
+						this.setState({
+							dndVariants: this.getOrderedShowStyleVariants(),
+						}),
+					timeout
+				)
+			}
+		}
+
+		componentWillUnmount() {
+			clearTimeout(timer)
+		}
+
+		private showStyleVariantsChanged = (
+			prevState: IShowStyleVariantsSettingsState,
+			prevProps: Readonly<Translated<IShowStyleVariantsProps>>
+		): boolean => {
+			if (this.props.showStyleVariants.length > 0 && this.state.dndVariants.length === 0) {
+				return true
+			}
+
+			if (prevState.dndVariants.length !== this.props.showStyleVariants.length) {
+				return true
+			}
+
+			return prevProps.showStyleVariants !== this.props.showStyleVariants && this.state.editedMappings.length > 0
```

Comment:
It has to be checked for the UI to update while you for example add a new table to the configuration. Otherwise it will not visibly show it, therefore to easily account for all things that happen during editing, the edited mappings can be used.

Reply:
And by adding a "new table" you mean adding a new ShowStyleVariant?

Reply:
I mean in the case where you have a variant that you click "edit" on, and then do changes. Such as adding a new configuration table, like the GFX Design Template, those changes aren't shown to the user without the `editedMappings` checked right now.

Reply:
That's... surprising...